### PR TITLE
[lldb] Guard SetDescriptorFinderAndClearOnExit against recursion

### PR DESCRIPTION
**Pull request by augusto2112**
If a function in ReflectionContext.cpp sets the descriptor finder and calls another one that does the same, when the second function finishes the descriptor finder will be cleared. Fix this by keeping a stack of descriptor finders.

rdar://122586257

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/ReflectionContext.cpp [hunk 1]
```diff
@@ -76,7 +80,7 @@ struct DescriptorFinderForwarder : public swift::reflection::DescriptorFinder {
     }
   }
 
-  swift::reflection::DescriptorFinder *m_descriptor_finder = nullptr;
+  llvm::SmallVector<swift::reflection::DescriptorFinder *> m_descriptor_finders;
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/ReflectionContext.cpp]: What's a good default size to use here? 1?

**Review by adrian-prantl — APPROVED** [review]

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test windows

**Pull request merged**